pkg/ccl/crosscluster/logical/logical_replication_writer_processor.go

```diff
@@ -674,6 +680,30 @@ func (lrw *logicalReplicationWriterProcessor) flushBuffer(
 		return a.KeyValue.Value.Timestamp.Compare(b.KeyValue.Value.Timestamp)
 	})
 
+	// If the seen map is nil or has hit 2M items, reset it.
+	if lrw.seenKeys == nil || len(lrw.seenKeys) > 2<<20 {
+		lrw.seenKeys = make(map[uint64]int64, 2<<20)
+	}
+
+	h := fnv.New64a()
+	logged := false
+	for i := range kvs {
```

Comment:
i'm not sure we want this extra loop over the kvs in all prod deployments. Should we gate this behind a verbose logging conditional or some dumb "ldr debug" cluster setting?

Reply:
Eh, this loop is directly below one which _sorts_ the slice, and above one where we'll do a whole RPC for every item in it, while we're just doing a couple xors. I'm no too inclined to worry about it until a profile says otherwise.

> behind a verbose logging conditional

Eh, we want to know if this is happening and we need to be doing the counting to know it if is.

Reply:
Sounds good. I don't feel too strongly about this. Also, should `key` get redacted in the log line?

Reply:
> what makes key different in this log line than span in the lag log line or row in the dlq log line?

obviously no difference. but we'll need to fix this eventually.